test/box-luatest/gh_10265_memtx_checkpoint_fail_assertion_test.lua

```diff
@@ -0,0 +1,28 @@
+local t = require('luatest')
+local server = require('luatest.server')
+
+local g = t.group()
+
+g.before_all(function(cg)
+    t.tarantool.skip_if_not_debug()
+    cg.server = server:new()
+    cg.server:start()
+end)
+
+g.after_all(function(cg)
+    cg.server:drop()
+end)
+
+g.test_memtx_checkpoint_fail_assertion = function(cg)
+    cg.server:exec(function()
+        local s = box.schema.space.create('test')
+        s:create_index('pk')
+        s:replace{0, 0}
+        -- Set error injection to iterator creation to fail
+        -- memtx checkpoint right from the start.
+        box.error.injection.set('ERRINJ_INDEX_ITERATOR_NEW', true)
```

Comment:
I doubt it's possible to reproduce this issue without an error injection so it doesn't make much sense to include the fix into the release notes - I'd drop the changelog.

Reply:
Yes, I think it's okay to merge this one.